File: kmp-migrate-room/shared/src/commonMain/kotlin/com/example/fruitties/kmptutorial/android/model/CartItem.kt
package com.example.fruitties.kmptutorial.android.model

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import androidx.room.Relation
import kotlin.experimental.ExperimentalObjCName
import kotlin.native.ObjCName

@OptIn(ExperimentalObjCName::class)
@ObjCName("CartItemEntity")
@Entity(
    foreignKeys = [
        ForeignKey(
            entity = Fruittie::class,
            parentColumns = ["id"],
            childColumns = ["id"],
            onDelete = ForeignKey.CASCADE,
        ),
    ],
)
data class CartItem(@PrimaryKey val id: Long, val count: Int = 1)

data class CartItemWithFruittie(
    @Relation(
        parentColumn = "id",
        entityColumn = "id",
    )
    val fruittie: Fruittie,
    @Embedded val cartItem: CartItem,
)

File: kmp-migrate-room/iosApp/Sources/Repository/CartItem.swift
import sharedKit

struct CartItem: Hashable {
   let entity: CartItemWithFruittie

   let fruittie: Fruittie?

   var id: Int64 {
       entity.cartItem.id
   }

   var count: Int64 {
       Int64(entity.cartItem.count)
   }

   init(entity: CartItemWithFruittie) {
       self.entity = entity
       self.fruittie = Fruittie(entity: entity.fruittie)
   }
}
